pkgs/development/cuda-modules/_cuda/lib/versions.nix

```
@@ -69,11 +66,59 @@ in
     ```
     :::
   */
-  trimComponents =
-    n: v:
-    lib.pipe v [
-      lib.splitVersion
-      (lib.take n)
-      (lib.concatStringsSep ".")
-    ];
+  trimComponents = n: v: lib.concatStringsSep "." (quantizeVersion n v);
+
+  /**
+    ```nix
+    quantizeVersion 2 "1.2.3"
+    => [ "1" "2" ]
+    ```
+  */
+  quantizeVersion = components: version: lib.take components (lib.versions.splitVersion version);
+
+  /*
+    ∷ VersionString → Bool
+
+    Test equality of versions quantized to highest common precision.
+
+    ```nix
+    matchQuantized "12" "12.2"
+    => true
+    ```
+
+    ```nix
+    matchQuantized "12" "11.8"
+    => false
+    ```
+  */
+  matchQuantized =
+    v1: v2:
+    let
+      components1 = lib.length (lib.versions.splitVersion v1);
+      components2 = lib.length (lib.versions.splitVersion v2);
+      components = lib.min components1 components2;
+    in
+    quantizeVersion components v1 == quantizeVersion components v2;
```

```suggestion
    let
      components1 = lib.versions.splitVersion v1;
      components2 = lib.versions.splitVersion v2;
      minNumComponents = lib.min (lib.length components1) (lib.length components2);
    in
    lib.take minNumComponents components1 == lib.take minNumComponents components2;
```

Wasteful to re-split the string.